### PR TITLE
to_chars libc++ compatibility

### PR DESCRIPTION
**Pull request by mordante**
@StephanTLavavej here some changes I made in libc++ for https://reviews.llvm.org/D70631 "Microsoft's floating-point to_chars powered by Ryu and Ryu Printf". If you don't like `_Output` feel free to suggest an alternative and I'll adjust libc++ accordingly.

### CODE REVIEW

**Comment by mordante** [comment]
I also notices line 1103 and 1105 look oddly similar in their suggestion "Consider retuning PrefixesToTest and FractionBits.". I suspect that's not intended. Both also have a small typo in "returning".

File: stl/inc/xcharconv_ryu.h
Comment (miscco) [inline stl/inc/xcharconv_ryu.h:1155]: I think here should be `_Output`

File: tests/std/tests/P0067R5_charconv/test.cpp
Comment (miscco) [inline tests/std/tests/P0067R5_charconv/test.cpp:104]: out of curriosity is there any goal to use these tests with libc++

File: stl/inc/xcharconv_ryu.h
Comment (mordante) [reply stl/inc/xcharconv_ryu.h]: Yeah thanks, seems I accidentally removed it.

File: tests/std/tests/P0067R5_charconv/test.cpp
Comment (mordante) [reply tests/std/tests/P0067R5_charconv/test.cpp:104]: STL created this PR https://reviews.llvm.org/D70631, including the test. The test already found one omission in libc++'s implementation and another oddity where I need to verify whether libc++ is standard conforming.

**Comment by StephanTLavavej** [comment]
> I also notices line 1103 and 1105 look oddly similar in their suggestion "Consider retuning PrefixesToTest and FractionBits.". I suspect that's not intended. Both also have a small typo in "returning".

https://github.com/microsoft/STL/blob/834baa6acacb8325367e255b0728c582c7e443ff/tests/std/tests/P0067R5_charconv/test.cpp#L1100-L1104

It was intentional, although I can see now how it could be confusing. By "retuning" I meant "re-tuning", and the similar messages were intended to say "if this test ran really fast, you should probably be testing more random values" and "if this test ran really slow, you should probably test fewer random values". I can clarify this.

**Review by StephanTLavavej — APPROVED** [review]
Looks good to me! I've pushed two small commits, one to restore indentation (clang-format is disabled for the Ryu-derived code so it can stay somewhat consistent with upstream, otherwise it would have detected the extra space) and another to clarify the fast/slow comments (also in another test that had used the same wording).

Our system involves 2 reviews before simultaneously merging to GitHub and the MSVC-internal repo, so we should be able to get a second reviewer and batch this up for merging early next week.

**Comment by mordante** [comment]
> Looks good to me! I've pushed two small commits, one to restore indentation (clang-format is disabled for the Ryu-derived code so it can stay somewhat consistent with upstream, otherwise it would have detected the extra space) and another to clarify the fast/slow comments (also in another test that had used the same wording).

The new wording looks clearer, thanks. 

> Our system involves 2 reviews before simultaneously merging to GitHub and the MSVC-internal repo, so we should be able to get a second reviewer and batch this up for merging early next week.

Great, once it has landed I'll sync the code in libc++.

**Review by CaseyCarter — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks for keeping our codebases aligned, and congratulations on your first microsoft/STL commit! :smile_cat: :tada: :fireworks:

**Pull request merged**
